tests/test_auto_annotator.py

```diff
@@ -0,0 +1,223 @@
+    [
+        ("ten dollars", "$", 10),
+        ("two dollars", "$", 2),
+        ("two hundred dollars", "$", 200),
+        ("$58.67", "$", 58.67),
+        ("thirty-eight euros", "EUR", 38),
+        ("1/4 dollar", "$", 0.25),
+        ("seventy-eight euros", "EUR", 78),
+        ("$70k", "$", 70000),
+    ],
+)
+def test_money_parse(get_SA, query, unit, value):
+    spacy_response = get_SA.parse(query)[0]
+    assert unit == spacy_response["value"]["unit"]
+    assert value == spacy_response["value"]["value"]
+    assert spacy_response["dim"] == "sys_amount-of-money"
+
+
+@pytest.mark.parametrize(
+    "query, unit, value",
+    [
+        ("about 2 hours", "hour", 2),
+        ("2 hours", "hour", 2),
+        ("half an hour", "minute", 30),
+        ("15 minutes", "minute", 15),
+        ("nearly 15 minutes", "minute", 15),
+        ("2 days", "day", 2),
+        ("4 years", "year", 4),
+        ("3 weeks", "week", 3),
+    ],
+)
+def test_duration_parse(get_SA, query, unit, value):
+    spacy_response = get_SA.parse(query)[0]
+    assert unit == spacy_response["value"]["unit"]
+    assert value == spacy_response["value"]["value"]
+    assert spacy_response["dim"] == "sys_duration"
+
+
+@pytest.mark.parametrize(
+    "query, grain, value",
+    [
+        ("March 3rd 2012", "day", "2012-03-03T00:00:00.000-08:00"),
+        ("Feb 18 2019", "day", "2019-02-18T00:00:00.000-08:00"),
+        ("first thursday of 2015", "day", "2015-01-01T00:00:00.000-08:00"),
+    ],
+)
+def test_time_parse(get_SA, query, grain, value):
+    spacy_response = get_SA.parse(query)[0]
+    assert grain == spacy_response["value"]["grain"]
+    assert value == spacy_response["value"]["value"]
+    assert spacy_response["dim"] == "sys_time"
+
+
+@pytest.mark.parametrize(
+    "rule, pattern",
+    [
+        (
+            {
+                "domains": "faq|salary",
+                "intents": "*",
+                "files": "train.txt|test.txt",
+                "entities": "sys_amt-of-money|sys_time",
+            },
+            ".*/(faq|salary)/.+/(train.txt|test.txt)",
+        ),
+        (
+            {
+                "domains": "salary",
+                "intents": "get_salary_aggregate|get_salary",
+                "files": "train.txt|test.txt",
+                "entities": "(sys_amt-of-money)",
+            },
+            ".*/salary/(get_salary_aggregate|get_salary)/(train.txt|test.txt)",
+        ),
+        (
+            {
+                "domains": "date",
+                "intents": "get_date",
+                "files": "train.txt|test.txt",
+                "entities": "sys_duration|sys_interval|sys_time",
+            },
+            ".*/date/get_date/(train.txt|test.txt)",
+        ),
+        (
+            {"domains": "general", "intents": "*", "files": "*", "entities": "*"},
+            ".*/general/.+/.+",
+        ),
+        (
+            {"domains": "*", "intents": "*", "files": "*", "entities": "*"},
+            ".*/.+/.+/.+",
```

Comment:
Add a case that uses the kleene star on something other than '.', e.g. "\d*" or "[A-Z]*". I think the current parser messes up.

Reply:
Done